tests/connect/test_spark_joins.py

```diff
@@ -43,7 +43,21 @@ def test_columns_after_join(make_spark_df):
 
     assert set(joined_df1.schema().column_names()) == set(["A", "B", "right.A"])
 
-    assert set(joined_df2.schema().column_names()) == set(["A", "B"])
+    assert set(joined_df2.schema().column_names()) == set(["A", "B", "right.A"])
```

Comment:
same here. We're bound to what spark expects. If our joins are returning additional columns, that could be very confusing for end users.